thunder/tests/distributed/test_ddp.py

```diff
@@ -28,6 +28,27 @@
 from thunder.tests.framework import TorchExecutor, nvFuserExecutor
 from thunder.tests.framework import instantiate
 
+# It is important to set this flag so that TE doesn't use
+# `torch.compile` to fuse a few operations. This is because
+# `torch.compile` creates a new process and that leads to
+# the error : daemonic processes are not allowed to have children
+# when running the tests.
+# With the setting below, we use `torch.jit` for this test suite
+# See: https://github.com/NVIDIA/TransformerEngine/blob/a38b291b0d1b04847e8ab1df8550df642a03a27d/transformer_engine/pytorch/jit.py#L11-L19
+os.environ["NVTE_TORCH_COMPILE"] = "0"
```

Comment:
It would be best to set this with pytest monkeypatch or unittest mock for the tests that run. This could otherwise become a source of future painful test interacitons

Reply:
This flag affects the behaviour at import time so we have to have it before the import

Also, this flag should only affect the TransformerEngine tests in this particular file as this flag would be unset at the end of test script.

https://stackoverflow.com/questions/68040747/why-does-environment-variable-get-deleted-after-the-end-of-a-python-script

Reply:
Yes, `NVTE_TORCH_COMPILE` is a TransformerEngine env variable. The issue is mainly that `torch.compile` doesn't work well with multiprocessing pool (see https://github.com/pytorch/pytorch/issues/97992). I will try to use `os.unsetenv` for now.

Reply:
My bad, unittest.mock did do the trick. Thanks @carmocca @tfogal